### PR TITLE
Create Initial Lambda Function

### PR DESCRIPTION
**Pull request by nelsonic**
This PR creates an AWS Lambda function that sends emails based on a template.

+ [x] We have shown that it can be invoked from an `Elixir` App: 
https://github.com/dwyl/elixir-invoke-lambda-example
+ [x] Save `event` to S3 for debugging #3 
+ [x] Create detailed setup instructions for creating SNS topic for SES notifications #1 
+ [x] Receive and _parse_ an SNS notification #1 
+ [x] Send the bare minimum data back to the Phoenix **`email`** App. https://github.com/dwyl/email
+ [x] Explain how to deploy the lambda function and what environment variables are needed.

Temporary (ascii) diagram: #14 (_help wanted_)

```
The main App does not do any Email as that is is not it's core function.
It delegates all email sending and monitoring activity to the aws-ses-lambda.

┌─────┐  send   ┌────────────────┐
| App | ───────>| aws-ses-lambda |─┐ The Lambda function Sends email
└─────┘  email  └────────────────┘ | and handles SNS notifications
                                   | for bounce events.
  ┌───────┐    SNS Notification    |
  | Email | <──────────────────────┘
  |  App  |
  └───────┘  
  The Email Stats App aggregates and visualises email stats.
  This allows us to be more data-driven in our communications.
  And understand exactly who is engaged with the app.

  This is not a full on  it's just an attempt to make email
  sending/monitoring separate so our App can focus on core features.
```

### CODE REVIEW

**Comment by nelsonic** [comment]
image

Thanks AWS ...

**Comment by nelsonic** [comment]
Going to writeup some SES Setup instructions https://github.com/dwyl/learn-amazon-web-services/issues/33
And return to this.

**Comment by nelsonic** [comment]
Need to update **`dpl`** to Node V12 so we can use it to _deploy_ this! https://github.com/dwyl/aws-lambda-deploy/issues/65

**Comment by nelsonic** [comment]
SES Setup instructions: https://github.com/dwyl/learn-amazon-web-services/pull/34 :shipit:

**Comment by nelsonic** [comment]
I've made good progress on this in the last couple of days. 
The Lambda function works to send email and we can _invoke_ it from Elixir.
I did a _complete_ writeup of my **`SPIKE`** in: https://github.com/dwyl/elixir-invoke-lambda-example
I am convinced this is a good route to go down.

But as I get to the stage of handling the SNS Notifications (_delivery, bounce, complaint, etc._),
I _really_ need a way of _storing_ this data.
In the past we have used S3 or DynamoDB to store data created/received/transformed by Lambdas,
but these are not great for querying and visualising. 

I'm now reaching the point where I want to save data _out_ of the Lambda function. 
That's a bit more work ... it will be worth it!

File: README.md [hunk 1]
```diff
@@ -1,8 +1,293 @@
-# aws-ses-bounce-checker :mailbox_with_mail:
-A tool to check if emails sent by SES have bounced.
+<div align="center">
```
Comment (nelsonic) [inline README.md:1]: Read the rendered version: https://github.com/dwyl/aws-ses-lambda/tree/init

**Comment by nelsonic** [comment]
Hi @SimonLab! This PR is finally at the point where it can be reviewed. 🙏 
The `.js` code is almost comically simple. see:  [`/lib`](https://codecov.io/gh/dwyl/aws-ses-lambda/tree/master/lib)👨‍💻 
The important part is getting it setup on AWS, hence the comprehensive instructions in [**`SETUP.md`**](https://github.com/dwyl/aws-ses-lambda/blob/init/SETUP.md)

I have already deployed the Lambda function and tested it from Elixir https://github.com/dwyl/elixir-invoke-lambda-example and the SNS (bounce notification) data is sent to: https://github.com/dwyl/email ✉️ ✅ 

Please review and if everything is clear, merge. 
Thanks! 👍

**Comment by nelsonic** [comment]
@SimonLab these extra commits have been for updating the API endpoint to `/api/sns` for clarity. 👍

File: README.md [hunk 2]
```diff
@@ -1,8 +1,315 @@
...
+module.exports = function send (event, callback) {
+  return sendemail(event.template, event, callback);
+};
+```
+
+Don't you just _love_ it when things are _that_ simple?! <br />
+All the data required for sending an email
+is received in the Lambda **`event`** object.
+
+The required keys in the `even` object are:
```
Comment (SimonLab) [inline README.md]:
```suggestion
The required keys in the `event` object are:
```

**Review by SimonLab — COMMENTED** [review]
Looks great :+1: . 
I've only spotted a small typo.

File: README.md (hunk 2 above)
Comment (nelsonic) [reply README.md]: Good catch! Thanks for fixing. ✅

**Comment by nelsonic** [comment]
Thanks for reviewing and fixing the typo @SimonLab 🥇 
If you're happy, please merge so the repo looks a bit better. :shipit:

**Comment by nelsonic** [comment]
**`Before`**: [github.com/dwyl/aws-ses-lambda/tree/c0d4ce654fe975f97f9217d970bf8b0141ab562d](https://github.com/dwyl/aws-ses-lambda/tree/c0d4ce654fe975f97f9217d970bf8b0141ab562d)
***`After`***: [github.com/dwyl/aws-ses-lambda/blob/60ceece7d59bb3428da55b471c55c06080705f2d](https://github.com/dwyl/aws-ses-lambda/blob/60ceece7d59bb3428da55b471c55c06080705f2d/README.md)

**Pull request merged**